packages/dev/docs/pages/react-spectrum/testing.mdx

```diff
@@ -0,0 +1,158 @@
+<!-- Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License. -->
+
+import {Layout} from '@react-spectrum/docs';
+export default Layout;
+
+---
+category: Concepts
+---
+
+# Testing
+
+This page describes how to test an application built with with React Spectrum, including how to 
+query the DOM tree for elements and simulate user interactions.
+
+## Introduction
+
+Running automated tests on your application helps ensure that it continues to work as expected over time.
+You can use testing tools like [Enzyme](https://enzymejs.github.io/enzyme/) or
+[React Testing Library](https://testing-library.com/docs/react-testing-library/intro) along with test runners like
+[Jest](https://jestjs.io) or [Mocha](https://mochajs.org) to test React Spectrum applications. These generally 
+work quite well out of the box but there are a few things to consider to ensure your tests are the best they
+can be.
+
+## Testing semantics
+
+Many testing libraries expect you to query for elements in the DOM tree. For example, you might have a test 
+that renders a login page, finds the username and password fields, and simulates filling them out and 
+submitting the form.
+
+Querying for these elements can be done in many ways. It's common to use CSS selectors for this, 
+but this can be problematic if you rely on implementation details of the component in these queries.
+For example, if you query by a CSS class name that may change in a future update of React Spectrum, your
+test will be brittle.
+
+This also applies if you query for internal elements within a React Spectrum component. The DOM structure
+that React Spectrum renders should be considered a black box, and it may change at any time, so you should
+avoid relying on it.
+
+The correct way to query for React Spectrum components and their internals is by semantics. React Spectrum
+components implement [ARIA patterns](https://www.w3.org/TR/wai-aria-practices-1.2/). ARIA is a W3C standard
+that specifies the semantics for many UI components. This is used to expose them to assistive technology
+like screen readers, but can also be used in tests to operate the application programmatically. These semantics
+are much less likely to change over time, and while other DOM nodes may be added or removed, and CSS classes
+changed, the semantics will stay stable.
```

Comment:
can we guarantee that they won't change? Maybe say *the semantics are more stable ?